src/Mount.cpp

```diff
@@ -3163,41 +3270,69 @@ void Mount::calculateStepperPositions(float raCoord, float decCoord, long &raPos
 /////////////////////////////////
 void Mount::calculateRAandDECSteppers(long &targetRASteps, long &targetDECSteps, long pSolutions[6]) const
 {
-    //LOGV3(DEBUG_MOUNT_VERBOSE,F("[MOUNT]: CalcSteppersPre: Current: RA: %s, DEC: %s"), currentRA().ToString(), currentDEC().ToString());
-    //LOGV3(DEBUG_MOUNT_VERBOSE,F("[MOUNT]: CalcSteppersPre: Target : RA: %s, DEC: %s"), _targetRA.ToString(), _targetDEC.ToString());
-    //LOGV2(DEBUG_MOUNT_VERBOSE,F("[MOUNT]: CalcSteppersPre: ZeroRA : %s"), _zeroPosRA.ToString());
-    //LOGV4(DEBUG_MOUNT_VERBOSE,F("[MOUNT]: CalcSteppersPre: Stepper: RA: %l, DEC: %l, TRK: %l"), _stepperRA->currentPosition(), _stepperDEC->currentPosition(), _stepperTRK->currentPosition());
-    DayTime raTarget = _targetRA;
+    LOGV3(DEBUG_MOUNT_VERBOSE, F("[MOUNT]: CalcSteppersPre: Current: RA: %s, DEC: %s"), currentRA().ToString(), currentDEC().ToString());
+    LOGV3(DEBUG_MOUNT_VERBOSE, F("[MOUNT]: CalcSteppersPre: Target : RA: %s, DEC: %s"), _targetRA.ToString(), _targetDEC.ToString());
+    LOGV2(DEBUG_MOUNT_VERBOSE, F("[MOUNT]: CalcSteppersPre: ZeroRA : %s"), _zeroPosRA.ToString());
+    LOGV4(DEBUG_MOUNT_VERBOSE,
+          F("[MOUNT]: CalcSteppersPre: Stepper: RA: %l, DEC: %l, TRK: %l"),
+          _stepperRA->currentPosition(),
+          _stepperDEC->currentPosition(),
+          _stepperTRK->currentPosition());
+    DayTime raTarget      = _targetRA;
+    Declination decTarget = _targetDEC;
 
     raTarget.subtractTime(_zeroPosRA);
-    //LOGV3(DEBUG_MOUNT_VERBOSE,F("[MOUNT]: CalcSteppersIn: Adjust RA by Zeropos. New Target RA: %s, DEC: %s"), raTarget.ToString(), _targetDEC.ToString());
+    LOGV3(DEBUG_MOUNT_VERBOSE,
+          F("[MOUNT]: CalcSteppersIn: Adjust RA by Zeropos. New Target RA: %s, DEC: %s"),
+          raTarget.ToString(),
+          _targetDEC.ToString());
 
-    float hourPos = raTarget.getTotalHours();
+    // Where do we want to move RA to?
+    float moveRA = raTarget.getTotalHours();
     if (!NORTHERN_HEMISPHERE)
     {
-        hourPos += 12;
+        moveRA += 12;
+    }
+
+    // Total hours of tracking-to-date
+    float trackedHours = (_stepperTRK->currentPosition() / _trackingSpeed) / 3600.0F;  // steps / steps/s / 3600 = hours
+    LOGV3(
+        DEBUG_MOUNT_VERBOSE, F("[MOUNT]: CalcSteppersIn: Tracked time is %l steps (%f h)."), _stepperTRK->currentPosition(), trackedHours);
+
+    // The current RA of the home position, taking tracking-to-date into account
+    float homeRA = _zeroPosRA.getTotalHours() + trackedHours;
+
+    // Delta between target RA and home position with a normalized range of -12 hr to 12 hr
+    float homeTargetDeltaRA = _targetRA.getTotalHours() - homeRA;
+    while (homeTargetDeltaRA > 12)
+    {
+        homeTargetDeltaRA = homeTargetDeltaRA - 24;
     }
+    while (homeTargetDeltaRA < -12)
+    {
+        homeTargetDeltaRA = homeTargetDeltaRA + 24;
+    }
+
     // Map [0 to 24] range to [-12 to +12] range
```

Comment:
Seems like we keep doing a lot of this, maybe a future improvement could abstract this kind of operation :smiley: